src/lib/ipfs.ts

```diff
@@ -0,0 +1,104 @@
+import promisify = require("es6-promisify");
+import axios from "axios";
+import * as IPFS from 'ipfs-mini';
+const bs58 = require('bs58');
+
+// github.com/github/fetch/issues/175
+function timeout(promise, ms) {
+  return new Promise(function(resolve, reject) {
+    setTimeout(function() {
+      reject(new Error("timeout"))
+    }, ms)
+    promise.then(resolve, reject)
+  })
+}
+
+const ipfsHashPrefix = '1220';
+// Stores a static snapshot of the proposal data from before the ipfs migration
+const legacyIpfsHash = 'QmNaTgnLUAWneQwifeeLZHc5N1k2wqFKSfWPRjTS9ZjTxZ';
```

Comment:
IPFS nodes don't talk to each other very reliably, might need to add this file to your IPFS node manually.. 😕